dataclients/kubernetes/ingressv1.go

```diff
@@ -311,6 +312,26 @@ func (ing *ingress) addSpecRuleV1(ic ingressContext, ru *definitions.RuleV1) err
 	return nil
 }
 
+// addSpecIngressTLSV1 is used to add TLS Certificates from Ingress resources. Certificates will be added
+// only if the Ingress rule host matches a host in TLS config
+func (ing *ingress) addSpecIngressTLSV1(ic ingressContext, ingtls *definitions.TLSV1) error {
+	for _, htls := range ingtls.Hosts {
+		for _, rules := range ic.ingressV1.Spec.Rules {
+			if htls == rules.Host {
+				err := addHostTLSCert(ic, ingtls.Hosts, ingtls.SecretName, ic.ingressV1.Metadata.Namespace)
+				if err != nil {
+					log.Errorf("failed using secret %s for %s TLS", ingtls.SecretName, htls)
+					return nil
+				}
+				log.Debugf("loaded tls certificate for %s", htls)
+				return nil
+			}
+		}
+	}
```

Comment:
What are these loops for? Why not just call `addHostTLSCert(ic, ingtls.Hosts, ingtls.SecretName, ic.ingressV1.Metadata.Namespace)`? If we want to be strict then we should call `addHostTLSCert` only with a list of hosts found in `rules` that are also in `tls` hosts.

Reply:
Simplified the lookup by getting a list of matching `hosts` in `tls` and `rules` and registering the matching hosts in the cert registry.